[Redis] Advanced Sync rules and documentation (#2082)
elastic/connectors: docs/sync-rules/redis.md

### Setting up the Redis server/cloud connector

See the [Developer guide](../../docs/DEVELOPING.md) for setting up connectors.

### Overview

Advanced Sync Rules help manage data in Redis server/cloud. They take the following parameters:

> 1. `database`: Database indice of Redis server/cloud. Type of value should be integer.

Reviewer: ```suggestion
1. `database`: Database index of Redis server/cloud. The type of value should be integer.
```

2. `key_pattern`: Pattern of the key for searching in Redis server/cloud

> 3. `_type`: Type of key(i.e. HASH, LIST, SET, STREAM, STRING, ZSET) in Redis server/cloud.

Reviewer: ```suggestion
3. `_type`: Type of key in Redis server/cloud. Supported values are HASH, LIST, SET, STREAM, STRING, ZSET
```

Make supported values explicit

I actually had doubts when I reviewed https://github.com/elastic/connectors/pull/2044, seeing a method param starting with `_`. I give it a pass because I see `_type` is also used in [scan_iter](https://redis-py.readthedocs.io/en/stable/commands.html#redis.commands.core.CoreCommands.scan_iter)

But we should not define `_type` as a rule field. Or we should also consider rename all the `_type` to `type` in https://github.com/elastic/connectors/pull/2044

Author: But `type` is a built-in in python. can we use something like `key_type` instead of using built-in then?

Reviewer: You can suffix it, `type_`

*NOTE*:

> - `key_pattern` or `_type` is not required but any one of them is required to pass.

Reviewer: I think this needs to be re-phrased. cc @leemthompo WDYT?

Reviewer 2: Agree!

### Advanced sync rules examples

> #### Fetch all keys starting with alpha.

Reviewer: ```suggestion
#### Fetch database records with keys starting with alpha.
```
Make it consistent
